src/compas/geometry/_intersections/mesh_intersections.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+from compas.geometry import Point
+from compas.geometry import length_vector
+from compas.geometry import subtract_vectors
+from compas.geometry import intersection_line_triangle
+from compas.geometry import intersection_segment_plane
+
+__all__ = [
+    'intersection_mesh_line',
+    'intersection_mesh_plane',
+    'mesh_vertices_to_points',
+]
+
+
+def intersection_mesh_line(mesh, line):
+    """Compute intersection between mesh faces and line. After one single intersection, stops searching for more.
+
+    Parameters
+    ----------
+    mesh : compas.datastructures.Mesh
+    line : compas.geometry.Line
+
+    Returns
+    -------
+    Point : compas.geometry.Point
+    """
+    for fkey in list(mesh.faces()):
+        vertex_keys = mesh.face_vertices(fkey)
+        if not vertex_keys:
+            continue
+        vertices = [mesh.vertex_attributes(vkey, 'xyz') for vkey in vertex_keys]
+        if len(vertex_keys) not in (3, 4):
+            continue
+
+        triangle = [vertices[0], vertices[1], vertices[2]]
+        intersection = intersection_line_triangle(line, triangle)
+        if intersection:
+            return Point(intersection[0], intersection[1], intersection[2])
+
+        if len(vertex_keys) == 4:
+            triangle_2 = [vertices[2], vertices[3], vertices[0]]
+            intersection_2 = intersection_line_triangle(line, triangle_2)
+            if intersection_2:
+                return Point(intersection_2[0], intersection_2[1], intersection_2[2])
+                
+    return None
+
+
+def intersection_mesh_plane(mesh, plane, tol=0.0001):
+    """Calculate the keys of the points of the intersection of a mesh with a plane
+
+    Parameters
+    ----------
+    mesh : compas.datastructures.Mesh
+    plane : compas.geometry.Plane
+
+    Returns
+    -------
+    intersections: list of points as keys from mesh
+    """
+    intersections = []
+    for u, v in list(mesh.edges()):
+        a = mesh.vertex_attributes(u, 'xyz')
+        b = mesh.vertex_attributes(v, 'xyz')
+        intersection = intersection_segment_plane((a, b), plane)
+        if not intersection:
+            continue
+        len_a_inters = length_vector(subtract_vectors(intersection, a))
+        len_a_b = length_vector(subtract_vectors(b, a))
+        t = len_a_inters / len_a_b
+        if t >= 1.0:
+            t = 1 - tol
+        elif t <= 0.0:
+            t = tol
+        intersection_key = mesh.split_edge(u, v, t=t, allow_boundary=True)
+        intersections.append(intersection_key)
+    return intersections
+
+
+def mesh_vertices_to_points(mesh, v_keys):
```

Comment:
This function doesn't have much functionality, and I don't see much point of adding it to the library.